packages/style-engine/style-engine.php

```diff
@@ -54,13 +57,36 @@ function wp_style_engine_get_styles( $block_styles, $options = array() ) {
 	$styles_output = array();
 
 	if ( ! empty( $parsed_styles['declarations'] ) ) {
-		$styles_output['css']          = WP_Style_Engine::compile_css( $parsed_styles['declarations'], $options['selector'] );
-		$styles_output['declarations'] = $parsed_styles['declarations'];
-		if ( ! empty( $options['context'] ) ) {
-			WP_Style_Engine::store_css_rule( $options['context'], $options['selector'], $parsed_styles['declarations'] );
+		$container        = $options['container'] ?? null;
+		$selector         = $options['selector'] ?? null;
+		$css_declarations = new WP_Style_Engine_CSS_Declarations( $parsed_styles['declarations'] );
+		$new_rule         = null;
+
+		// TODO: Extend WP_Style_Engine::compile_css to do this if block.
+		if ( $selector || $container ) {
+			if ( $selector ) {
+				$new_rule = new WP_Style_Engine_CSS_Rule( $options['selector'], $css_declarations );
+			}
+
+			if ( $container ) {
+				$new_rule = new WP_Style_Engine_CSS_Rules_Container( $container, $new_rule );
```

Comment:
A `WP_Style_Engine_CSS_Rules_Container` also allows us to limit the depth of nested rules to 1